"Iterator[BodyPartReader]" has no attribute "__anext__"

🐞 **Describe the bug**
<!-- A clear and concise description of what the bug is, on the next line. -->
`BodyPartReader.__aiter__()` is declared to return `Iterator` instead of `AsyncIterator`. Introduced in https://github.com/aio-libs/aiohttp/commit/3b70bad264da8e47974264863a09d6a5fc808b39.


💡 **To Reproduce**
<!-- How to reproduce the behavior?

For example:
1. Have certain environment
2. Run given code snippet in a certain way
3. See some behavior described

Add these steps below this comment: -->

Snippet to reproduce:

```Python
# rep.py
from aiohttp import web

async def handler(request: web.Request) -> None:
    reader = await request.multipart()
    async for field in reader:
        ...
```


💡 **Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->

No errors returned by mypy.

📋 **Logs/tracebacks**
<!-- If applicable, add logs/tracebacks to help explain your problem. -->
```python-traceback (paste your traceback in the next line)
$ mypy rep.py
rep.py:5: error: "Iterator[BodyPartReader]" has no attribute "__anext__"; maybe "__next__"?
Found 1 error in 1 file (checked 1 source file)
```

📋 **Your version of the Python**
<!-- Attach your version of the Python. -->
```console
$ python --version
Python 3.8.6
```

📋 **Your version of the aiohttp/yarl/multidict distributions**
<!-- Attach your version of the distributions in the code blocks below. -->
```console
$ python -m pip show aiohttp
Name: aiohttp
Version: 3.7.2
Summary: Async http client/server framework (asyncio)
Home-page: https://github.com/aio-libs/aiohttp
Author: Nikolay Kim
Author-email: fafhrd91@gmail.com
License: Apache 2
Location: /private/tmp/aiohttpbug/.venv/lib/python3.8/site-packages
Requires: multidict, chardet, yarl, async-timeout, typing-extensions, attrs
Required-by:
```
```console
$ python -m pip show multidict
# irrelevant
```
```console
$ python -m pip show yarl
# irrelevant
```

📋 **Additional context**
<!-- Add any other context about the problem here, in the next line. -->

<!-- Describe the environment you have that lead to your issue.
     This includes aiohttp version, OS, proxy server and other bits that
     are related to your case.

     IMPORTANT: aiohttp is both server framework and client library.
     For getting rid of confusing please put 'server', 'client' or 'both'
     word here.
     -->

--- a/CHANGES/5163.bugfix
+++ b/CHANGES/5163.bugfix
@@ -1,0 +1,1 @@
+Change return type of MultipartReader.__aiter__() and BodyPartReader.__aiter__() to AsyncIterator.

--- a/CONTRIBUTORS.txt
+++ b/CONTRIBUTORS.txt
@@ -174,6 +174,7 @@ Kimmo Parviainen-Jalanko
 Kirill Klenov
 Kirill Malovitsa
 Konstantin Valetov
+Krzysztof Blazewicz
 Kyrylo Perevozchikov
 Kyungmin Lee
 Lars P. Søndergaard

--- a/aiohttp/multipart.py
+++ b/aiohttp/multipart.py
@@ -10,6 +10,7 @@ from types import TracebackType
 from typing import (
     TYPE_CHECKING,
     Any,
+    AsyncIterator,
     Dict,
     Iterator,
     List,
@@ -271,7 +272,7 @@ class BodyPartReader:
         self._content_eof = 0
         self._cache = {}  # type: Dict[str, Any]
 
-    def __aiter__(self) -> Iterator["BodyPartReader"]:
+    def __aiter__(self) -> AsyncIterator["BodyPartReader"]:
         return self  # type: ignore
 
     async def __anext__(self) -> bytes:
@@ -583,7 +584,7 @@ class MultipartReader:
 
     def __aiter__(
         self,
-    ) -> Iterator["BodyPartReader"]:
+    ) -> AsyncIterator["BodyPartReader"]:
         return self  # type: ignore
 
     async def __anext__(

